### ISSUE TITLE
Review HloInstruction::CreateFromProto and double-check that we haven't missed anything

### ISSUE THREAD

**Issue by burmako** [openxla/stablehlo#565]
Recently, I came across [HloInstruction::CreateFromProto](https://github.com/tensorflow/tensorflow/blob/1055588dd3ea6a5b4574f60dbf797244f4a3065a/tensorflow/compiler/xla/hlo/ir/hlo_instruction.cc#L83), and I think that it provides a very nice summary of which HLO opcodes support which metadata.

In [HloInstructionProto](https://github.com/tensorflow/tensorflow/blob/1055588dd3ea6a5b4574f60dbf797244f4a3065a/tensorflow/compiler/xla/service/hlo.proto#L94), all kinds of metadata are packed into the same proto message, and the [Operation Semantics](https://www.tensorflow.org/xla/operation_semantics) page is not 100% up to date, so it was nice to discover an authoritative resource on the matter.

We should review this method and double-check if we missed anythin when speccing StableHLO.

**Comment by burmako** [comment]
For example, here's one thing that we missed: [kParameter::parameter_replicated_at_leaf_buffers](https://github.com/tensorflow/tensorflow/blob/99c6719d61863c40f43e84e81e55eed1a41b3278/tensorflow/compiler/xla/hlo/ir/hlo_instruction.cc#L452-L459).

**Comment by burmako** [comment]
I've reviewed all the ops that we have already specced and found only a few gaps mentioned above in the GitHub UI in referenced tickets. None of these gaps look significant, which is great - this means that speccing is going very well.

The rest of the ops will be reviewed as we go according to [the checklist](https://github.com/openxla/stablehlo/blob/main/docs/spec_checklist.md), which is proposed to be expanded with this procedure in #625.
